src/test/java/com/cgdpd/library/service/BookServiceTest.java

```diff
@@ -95,6 +104,86 @@ public void should_throw_exception_if_author_not_exist() {
         verifyNoInteractions(bookRepository);
     }
 
+    @Test
+    void should_return_books_with_pagination() {
+        // given
+        var searchCriteria = SearchBookCriteria.builder().build();
+        var spec = byBookSearchCriteria(searchCriteria);
+        var paginationCriteria = PaginationCriteria.builder()
+              .pageIndex(0)
+              .pageSize(10)
+              .sort(Optional.of(new SortParams("title", SortParams.Direction.ASC)))
+              .build();
+        var pageRequest = paginationCriteria.toPageRequest();
+        var books = List.of(aBookEntity().title("The Adventurous").build(),
+              aBookEntity().title("Finder").build(),
+              aBookEntity().title("Killer").build());
+
+        given(bookRepository.findAll(spec, pageRequest))
+              .willReturn(new PageImpl<>(books));
+
+        // when
+        var result = bookService.findBooks(paginationCriteria,
+              searchCriteria);
+
+        // then
+        assertThat(result.content()).containsExactlyInAnyOrder(books.stream()
+              .map(bookMapper::mapToDetailedBookDto)
+              .toArray(DetailedBookDTO[]::new));
+    }
+
+    @Test
+    void should_return_empty_page_when_no_books_found() {
+        // given
+        var searchCriteria = SearchBookCriteria.builder().build();
+        var spec = byBookSearchCriteria(searchCriteria);
+        var paginationCriteria = PaginationCriteria.builder()
+              .pageIndex(0)
+              .pageSize(10)
+              .sort(Optional.of(new SortParams("title", SortParams.Direction.ASC)))
+              .build();
+        var pageRequest = paginationCriteria.toPageRequest();
+
+        Page<BookEntity> emptyPage = Page.empty();
+
+        given(bookRepository.findAll(spec, pageRequest)).willReturn(emptyPage);
+
+        // when
+        var resultPagedResponse = bookService.findBooks(paginationCriteria,
+              searchCriteria);
+
+        // then
+        assertThat(resultPagedResponse.content()).isEmpty();
+        assertThat(resultPagedResponse.pageNumber()).isEqualTo(0);
+        assertThat(resultPagedResponse.pageSize()).isEqualTo(10);
+        assertThat(resultPagedResponse.totalElements()).isEqualTo(emptyPage.getTotalElements());
+    }
+
+    @Test
+    void should_return_last_page() {
```

Comment:
Let's remove this test, since we are mocking the repository it doesn't make much sense for this, we'll do it in functional tests, when we implement the endpoint